src/main/java/org/apache/flink/benchmark/functions/LongNewSource.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.benchmark.functions;
+
+import org.apache.flink.api.connector.source.Boundedness;
+import org.apache.flink.api.connector.source.SplitEnumerator;
+import org.apache.flink.api.connector.source.SplitEnumeratorContext;
+import org.apache.flink.api.connector.source.lib.NumberSequenceSource;
+import org.apache.flink.api.connector.source.lib.util.IteratorSourceEnumerator;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * The source should produce same records as {@link LongSource}.
+ */
+public class LongNewSource extends NumberSequenceSource {
+	private final Boundedness boundedness;
+	private final long maxValue;
+
+	public LongNewSource(Boundedness boundedness, long maxValue) {
+		super(-1, -1); // we do not use the from/to of the underlying source
```

Comment:
What's wrong with `NumberSequenceSource`? Can not we use it as it is?

Reply:
I wanted to have exact same logic as the `LongSource`. `LongSource` generates records from 0 to maxValue for every parallel instance.

The `NumberSequenceSource` would split the range 0 to `maxValue` between all subtasks.

Reply:
As a side comment. I tried reusing only the reader and enumerator, but the problem is that serializers for splits and enumerator checkpoints are private.

Reply:
👍